src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,132 @@
+#include "aikido/constraint/SequentialSampleable.hpp"
+
+#include <dart/common/StlHelpers.hpp>
+
+namespace aikido {
+namespace constraint {
+
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>>& generators)
+    : mStateSpace(std::move(stateSpace))
+    , mGenerators(generators)
+    , mIndex(0)
+  {
+    for (std::size_t i = 0; i < mGenerators.size(); ++i)
+      assert(mGenerators[i]->getStateSpace() == mStateSpace);
+  }
+
+  // Documentation inherited
+  statespace::StateSpacePtr getStateSpace() const override
+  {
+    return mStateSpace;
+  }
+
+  // Documentation inherited
+  bool sample(statespace::StateSpace::State* state)
+  {
+    while (canSample())
+    {
+      if (mGenerators[mIndex]->canSample())
+        return mGenerators[mIndex]->sample(state);
+
+      mIndex++;
+    }
+    return false;
```

Comment:
This logic (i.e., "try the current generator to sample and return the result") is possible to return false even when one of the remaining generators can sample, which I don't think that's the expected behavior of `sample`.

I propose changing this logic to something like:
```python
index2 = index
for generator in generators[index2:]:
  result = generator.sample(state);
  index += 1
  if result:
    return True
return False
```
so that it tries to sample until it can sample and returns false when all the remaining generators are not able to draw a sample.

Reply:
Ah. Got it. 
Your proposal will account for `sample() may transiently fail, i.e. return false, at any point before then`